source/delegate/contracts/Delegate.sol

```diff
@@ -359,13 +363,23 @@ contract Delegate is IDelegate, Ownable {
 
     Rule memory rule = rules[senderToken][signerToken];
 
+    uint256 senderParam = rule.maxSenderAmount;
+
     // Ensure that a rule exists.
-    if(rule.maxSenderAmount > 0) {
+    if (senderParam > 0) {
```

Comment:
give this is duplicate code, put it in a pure/ internal function?

Reply:
Yeah I thought about that as I copied and pasted it lol... I'll do it now

Reply:
can you do this for the other pair, there is duplicate code for 
provdeOrder + getSenderQuote as well.

Reply:
Its just a single line of code though for that one - do you think its worth it?

Reply:
No, not for solidity. Just because of how solidity does internal functions. In other cases, I see value in the sense of reducing overhead to finding out pricing logic and debugging pricing issues.